support-frontend/app/services/stepfunctions/SupportWorkersClient.scala

```diff
@@ -44,7 +44,7 @@ case class CreateSupportWorkersRequest(
   emailGiftRecipient: Option[String],
   product: ProductType,
   firstDeliveryDate: Option[LocalDate],
-  paymentFields: PaymentFields,
+  paymentFields: Option[PaymentFields],
```

Comment:
making things optional is quite low level and doesn't necessarily explain why things are optional and in which combination - if we want more expressiveness consider rolling your own sum/product types to lock it down better.
However I realise this is a low level model as it's used for going between the lambdas, so it would complicate the serialisers or require a separate low level model.
Also theres a risk of going full membership-common although I think that was mostly confusing down to the implicit chains!
```scala
// example, this would replace `product: ProductType and paymentFields`
sealed trait TheProduct
case class Corporate extends TheProduct
sealed abstract class PaidProduct(paymentFields: PaymentFields) extends TheProduct
case class ....(paymentFields: PaymentFields) extends PaidProduct
```
See what you think - I am probably slightly in favour of keeping it low level for ease of serialisation but it depends what kind of bugs we are expecting.

Reply:
I did consider that actually because it would have made the changes quite a bit less extensive but I felt that using an option was as explicit and a bit more standard - it's obvious to any Scala dev at a single glance that it might not be there whereas a custom model takes a little bit more investigation to understand. Also we have `PaymentFields` in the first two lambdas and `PaymentMethod` after that (when the fields have been converted into a payment method) so it would have meant doing the same thing for both classes which is starting to get a bit too elaborate for the benefit I think.

Reply:
> obvious to any Scala dev at a single glance that it might not be there

yes I agree there, certainly a step above nulls,, but I was thinking both telling and enforcing a why/when it might not be there rather than making it easy to see that it may or may not be there.  Just to make Direct subs with no payment details a compiler error rather than a mess in zuora (in the true spirit of ADTs)
In the last week I've become slightly more in favour of elaborate types over elaborate repairs to zuora 🙈 
But happy to leave it as is, we have to balance added understanding time for everyone, over the smaller risk of something bad happening. 👍